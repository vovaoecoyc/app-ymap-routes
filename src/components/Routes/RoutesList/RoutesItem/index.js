import React from 'react';

import { AppContext } from '../../../../Context';
import styles from './RoutesItem.module.css';

class RoutesItem extends React.Component {
  static contextType = AppContext;

  handlerClick = () => {
    this.context.removePoint(this.props.id);
  };

  render() {
    let {
      point: { name },
    } = this.props;
    return (
      <div className={`${styles.PointBlock}`}>
        <span>{name}</span>
        <div onClick={this.handlerClick} className={`${styles.RemoveIcon}`}>
          X
        </div>
      </div>
    );
  }
}

export default RoutesItem;
