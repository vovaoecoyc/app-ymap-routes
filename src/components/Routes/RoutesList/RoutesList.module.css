.RoutesListBlock {
  display: flex;
  flex-direction: column;
  margin-top: 20px;
}
.RoutesListBlock > div:not(:first-child) {
  margin-top: 10px;
}