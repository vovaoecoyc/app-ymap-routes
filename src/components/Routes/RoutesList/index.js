import React from 'react';

import RoutesItem from './RoutesItem';
import styles from './RoutesList.module.css';

class RoutesList extends React.Component {
  render() {
    const { points } = this.props;
    return (
      <div className={`${styles.RoutesListBlock}`}>
        {points.map((value, i) => (
          <RoutesItem key={i} id={i} point={value} />
        ))}
      </div>
    );
  }
}

export default RoutesList;
