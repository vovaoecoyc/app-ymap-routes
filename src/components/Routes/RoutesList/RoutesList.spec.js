import React from 'react';
import { shallow } from 'enzyme';
import RoutesList from '../RoutesList';

describe('RoutesList container', () => {
  const props = {
    points: [
      { coordinates: [55.1, 38.1], name: 'Name Point1' },
      { coordinates: [55.9, 38.0], name: 'Name Point2' },
    ],
  };

  const containerRoutesList = shallow(<RoutesList {...props} />);

  it('RoutesList initial snapshot', () => {
    expect(containerRoutesList).toMatchSnapshot();
  });

  it('Is render RoutesItem components', () => {
    expect(containerRoutesList.find('RoutesItem')).toHaveLength(props.points.length);
  });
});
