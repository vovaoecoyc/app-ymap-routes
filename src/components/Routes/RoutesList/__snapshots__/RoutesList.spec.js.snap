// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutesList container RoutesList initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoutesList
    points={
      Array [
        Object {
          "coordinates": Array [
            55.1,
            38.1,
          ],
          "name": "Name Point1",
        },
        Object {
          "coordinates": Array [
            55.9,
            38,
          ],
          "name": "Name Point2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RoutesItem
          id={0}
          point={
            Object {
              "coordinates": Array [
                55.1,
                38.1,
              ],
              "name": "Name Point1",
            }
          }
        />,
        <RoutesItem
          id={1}
          point={
            Object {
              "coordinates": Array [
                55.9,
                38,
              ],
              "name": "Name Point2",
            }
          }
        />,
      ],
      "className": "RoutesListBlock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "id": 0,
          "point": Object {
            "coordinates": Array [
              55.1,
              38.1,
            ],
            "name": "Name Point1",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "id": 1,
          "point": Object {
            "coordinates": Array [
              55.9,
              38,
            ],
            "name": "Name Point2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RoutesItem
            id={0}
            point={
              Object {
                "coordinates": Array [
                  55.1,
                  38.1,
                ],
                "name": "Name Point1",
              }
            }
          />,
          <RoutesItem
            id={1}
            point={
              Object {
                "coordinates": Array [
                  55.9,
                  38,
                ],
                "name": "Name Point2",
              }
            }
          />,
        ],
        "className": "RoutesListBlock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "id": 0,
            "point": Object {
              "coordinates": Array [
                55.1,
                38.1,
              ],
              "name": "Name Point1",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "id": 1,
            "point": Object {
              "coordinates": Array [
                55.9,
                38,
              ],
              "name": "Name Point2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
